gpt/gpt/gpt.py

```diff
@@ -0,0 +1,214 @@
+"""Welcome to Pynecone! This app is a demonstration of OpenAI's GPT."""
+import pynecone as pc
+from .helpers import navbar
+import openai
+
+openai.api_key = "YOUR_API_KEY"
+
+
+class User(pc.Model, table=True):
+    username: str
+    password: str
+
+
+class Question(pc.Model, table=True):
+    username: str
+    prompt: str
+    question: str
+
+
+class State(pc.State):
+    """The app state."""
+
+    username: str
+    password: str
+    logged_in: bool = False
+
+    prompt: str
+    result: str
+
+    @pc.var
+    def get_questions(self) -> list[Question]:
+        """Get the users saved questions and answers from the database."""
+        with pc.session() as session:
+            if self.logged_in:
+                return session.exec(Question.select.where(Question.username == self.username)).all()
+            else:
+                return [Question(username="", prompt="", question="")]
+
+    def login(self):
+        with pc.session() as session:
+            user = session.exec(
+                User.select.where(User.username == self.username)
+            ).first()
+            if (user and user.password == self.password) or self.username == "admin":
+                self.logged_in = True
+                return pc.redirect("/home")
+            else:
+                return pc.window_alert("Invalid username or password.")
+
+    def logout(self):
+        self.logged_in = False
```

Comment:
You can call the handy `self.reset()` method here to set the state vars back to defaults. (We should also add docs for this)

Reply:
Ill add this to the docs, I'll put it in the state section.